server/src/utils/testUtils.ts

```diff
@@ -0,0 +1,56 @@
+import { loggedInAgent } from '../../tests/setuptests';
+import Roadmap from '../api/roadmaps/roadmaps.model';
+import Customer from '../api/customer/customer.model';
+import User from '../api/users/users.model';
+import { Role } from '../api/roles/roles.model';
+import TaskRating from '../api/taskratings/taskratings.model';
+import { Permission, RoleType } from '../../../shared/types/customTypes';
+
+export const removePermission = async (permission: Permission) => {
+  const firstRoadmapId = (await Roadmap.query().first()).id;
+  const userId = (
+    await User.query().where({ username: 'AdminPerson1' }).first()
+  ).id;
+
+  await Role.query().patchAndFetchById([userId, firstRoadmapId], {
+    type: RoleType.Admin & ~permission,
+  });
```

Comment:
Just noticed...

This should really use the previous permission to remove from, not the Admin permission.

Now this will not work as intended when called multiple times to remove multiple permissions.

Reply:
I completely missed that. Would this change make sense? Now the function would take a RoleType as a parameter and the function caller could remove as many permissions from the original role as they want. Caller could also remove permissions from any role (dev, business etc.), not just admin.

Function:
```
export const removePermission = async (restrictedRole: RoleType) => {
  const firstRoadmapId = (await Roadmap.query().first()).id;
  const userId = (
    await User.query().where({ username: 'AdminPerson1' }).first()
  ).id;

  await Role.query().patchAndFetchById([userId, firstRoadmapId], {
    type: restrictedRole,
  });
};
```

Usage:
```
await removePermission(
   RoleType.Admin & ~Permission.RoadmapReadUsers & ~Permission.TaskRead,
);
```

From what I've tested, it seems to work well. I just figured it would be better to get it peer-reviewed before going around every file and change them to use this. I made a temporary fixup commit where the function is called `removePermission2` and it's used in `tests/tasks.test.ts -> GET -> Should not get tasks with incorrect permissions` if you want to try the functionality.